src/NostrIgnition.ts

```diff
@@ -401,79 +381,64 @@ const openNewWindow = (url: string): void => {
 };
 
-const connect = async (remotePubkey: string): Promise<Nip46Response | void> => {
-    console.log("Connecting to bunker...");
-    return nip46
-        .connect(remotePubkey)
-        .then((response) => {
-            if (response.result === "auth_url" && response.error) {
-                openNewWindow(`${response.error}?redirect_uri=${options.redirectUri}`);
-            }
-            return response;
-        })
-        .catch((error) => console.error(error));
-};
-
-const createAccount = async (
-    bunkerPubkey: string,
+const create = async (
+    bunkerProfile: BunkerProfile,
     username: string,
     domain: string,
-    email?: string
-): Promise<Nip46Response | void> => {
-    console.log("Creating account...");
-    return nip46
-        .createAccount(bunkerPubkey, username, domain, email)
-        .then((response) => response)
+    email?: string | undefined
+): Promise<string> => {
+    return createAccount(bunkerProfile, bunkerSignerParams, username, domain, email).then((newBunker) => {
+        bunker = newBunker;
+        console.log("Account created with pubkey: ", bunker.remotePubkey);
+        hasConnected = true;
+        resetForms();
+        return bunker.remotePubkey;
+    });
+};
+
+const connect = async (): Promise<void> => {
+    console.log("Connecting to bunker...");
+    return bunker
+        .connect()
+        .then(() => {
+            console.log("Connected to bunker!");
+            hasConnected = true;
+            resetForms();
+        })
         .catch((error) => console.error(error));
 };
 
-const ping = async (): Promise<Nip46Response | void> => {
+const ping = async (): Promise<void> => {
     console.log("Pinging bunker...");
-    nip46
+    bunker
         .ping()
-        .then((response) => {
-            if (response.result === "pong") {
-                console.log("Pong!");
-            } else if (response.result === "auth_url" && response.error) {
-                openNewWindow(`${response.error}?redirect_uri=${options.redirectUri}`);
-            }
-            return response;
-        })
+        .then(() => console.log("Pong!"))
         .catch((error) => console.error(error));
 };
 
-const signEvent = async (event: UnsignedEvent): Promise<Nip46Response | void> => {
+const signEvent = async (event: UnsignedEvent): Promise<VerifiedEvent | void> => {
     console.log("Requesting signature...");
-    return nip46
-        .sign_event(event)
-        .then((response) => {
-            if (response.result === "auth_url" && response.error) {
-                openNewWindow(`${response.error}?redirect_uri=${options.redirectUri}`);
-            } else {
-                console.log("Signed event:", JSON.parse(response.result));
-            }
-            return response;
-        })
-        .catch((error) => {
-            console.error(error);
-        });
+    return bunker
+        .signEvent(event)
+        .then((signedEvent) => console.log("Event signed!", signedEvent))
```

Comment:
You forgot to return the actual signed event here. This will just return `void`.

Reply:
oops.